java/arcs/core/storage/ReferenceModeStore.kt

```diff
@@ -183,25 +173,17 @@ class ReferenceModeStore private constructor(
     override suspend fun idle() {
         backingStore.idle()
         containerStore.idle()
+        receiveQueue.flush().join()
     }
 
     override fun on(
         callback: ProxyCallback<RefModeStoreData, RefModeStoreOp, RefModeStoreOutput>
     ): Int = callbacks.register(callback)
```

Comment:
Seems there was a bug that we missed flipping `callbacksHasBecomeEmpty` or similar after `callbacks.register(callback)`.

Reply:
Yes, this should be here, I don't know how I forgot that.